File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.cpp
class Solution {
public:
    void setZeroes(vector<vector<int>>& matrix) {
        vector<vector<int>> res(matrix);
        int m = matrix.size();
        int n = matrix[0].size();
        for(int i=0;i<m;i++){
            for(int j=0;j<n;j++){
                if(res[i][j] != 0)
                    res[i][j] = matrix[i][j];
                if(matrix[i][j] == 0){
                    for(int k=0;k<n;k++){
                        res[i][k] = 0;
                    }
                    for(int k=0;k<m;k++){
                        res[k][j] = 0;
                    }
                }
            }
        }
        matrix = res;
    }
};

File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.java
class Solution {
    public void setZeroes(int[][] matrix) {
        int m = matrix.length;
        int n = matrix[0].length;
        List<Integer> row = new ArrayList<>();
        List<Integer> col = new ArrayList<>();
        for(int i=0;i<m;i++){
            for(int j=0;j<n;j++){
                if(matrix[i][j] != 0) continue;
                if(!row.contains(i)) row.add(i);
                if(!col.contains(j)) col.add(j);
            }
        }
        for(int i=0;i<row.size();i++){
            int r = row.get(i);
            for(int j=0;j<n;j++){
                matrix[r][j] = 0;
            }
        }
        for(int i=0;i<col.size();i++){
            int c = col.get(i);
            for(int j=0;j<m;j++){
                matrix[j][c] = 0;
            }
        }
    }
}